var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -721,7 +721,7 @@ def configure_options(self):
                 hip_ipkgs.extend(["rocrand"])
             else:
                 hip_lpkgs.extend(["rocrand"])
-            if spec.satisfies("^hipblas@6.3.0:"):
+            if spec.satisfies("^hipblas-common"):
```

Comment:
I'll note - this change was added per discussion at https://github.com/spack/spack/pull/48644#discussion_r1921928015

cc: @alalazo

Reply:
BTW: I'm ok with this change/revert

Reply:
Actually thinking about it - the current code did flag an error - i.e. a misconfigured spec file for rocm-6.3

And the change would mask this and can trigger errors during build of petsc sources for this use case? [it will work for a rocm install where everything in in a single prefix/location though - but not when distributed - as when installed via spack]

Reply:
I was suggesting what is done in this PR, tbh. 

`petsc` declares dependencies on `hipblas`, not `hipblas-common`. So, checking whether `hipblas-common` is there means checking on a possibly transitive dependency: something on which the package has less control than a direct dependency.

Reply:
Okay, thanks @alalazo. Are you okay with this fix @balay, or is there something else you would like tested first?

Reply:
@alalazo I'm a bit confused.

> I was suggesting what is done in this PR

Hm - I had this code initially - and changed per prior discussion
```diff
 -           if spec.satisfies("^hipblas-common@6.3.0:"):
 +           if spec.satisfies("^hipblas@6.3.0:"):
```
This MR is reverting this change.

>  checking whether hipblas-common is there means checking on a possibly transitive dependency: something on which the package has less control than a direct dependency.

Are you suggesting a check for `transitive dependency` here is ok?  [its not clear]

Reply:
> I guess the fix here would be also list hipblas-common in external spec

Hm - maybe this wont' work [as it didn't work before for you - with the current code]

Reply:
@balay The other thing that is wrong in the previous PR (notice I didn't review it after the comment I left) is that you just changed the conditional, but you still refer to `hipblas-common` in the block after the conditional. 

Imo in a recipe you _can't_ assume to reach dependencies that you don't declare.

Reply:
So TL;DR, since we do:
```
spec[pkg]
```
for each `pkg in hip_ipkgs` and for each `pkg in hip_lpkgs`, imo we need to ensure these packages are also required dependencies.

Reply:
Just added an explicit dependency on `hipblas-common`, and tested it works okay for my case (added external `hipblas` and `hipblas-common` packages). Are you okay with this solution @alalazo and @balay?

Reply:
Fine with me. I'll let @balay merge the PR.